st2common/st2common/util/param.py

```diff
@@ -76,8 +76,7 @@ def _create_graph(action_context):
     Creates a generic directed graph for depencency tree and fills it with basic context variables
     '''
     G = nx.DiGraph()
-    G.add_node(SYSTEM_SCOPE, value=KeyValueLookup(scope=SYSTEM_SCOPE))
-    system_keyvalue_context = {SYSTEM_SCOPE: KeyValueLookup(scope=SYSTEM_SCOPE)}
+    system_keyvalue_context = {SYSTEM_SCOPE: KeyValueLookup(scope=FULL_SYSTEM_SCOPE)}
     G.add_node(DATASTORE_PARENT_SCOPE, value=system_keyvalue_context)
```

Comment:
I assume you mean why we don't do FULL_SYSTEM_SCOPE: KeyValueLookup(scope=FULL_SYSTEM_SCOPE)? 

This has to do with Jinja and this graph rendering code. If you see line 80, DATASTORE_PARENT_SCOPE is `st2kv`. So the key for the dictionary inside should just be `system` and not `st2kv.system`. In object notations, what we want is

```
st2kv: { 
   system: KeyValueLookup(scope=st2kv.system)
}
```
When we want to support user scoped variables resolution (which will be done after this PR gets merged), we'd have

```
st2kv: { 
   system: KeyValueLookup(scope=st2kv.system)
   user: KeyValueLookup(scope=st2kv.user)
}
```